src/environmentd/src/http.rs

```diff
@@ -726,16 +746,31 @@ async fn init_ws(
                 let creds = Credentials::Token { token };
                 (creds, options)
             }
+            WebSocketAuth::OptionsOnly { options: _ } => {
+                anyhow::bail!("Missing auth information!");
+            }
+        };
+        (auth(frontegg, creds).await?, options)
+    } else if let Some(user) = existing_user {
```

Comment:
This 4 branch if block is now really hard to reason about, especially because it depends on the state of 3 things (frontegg, ws_auth, existing_user), which makes it very easy to miss states and figure out if they should be problematic or not. For example, what happens if both frontegg and existing_user are specified? That should be impossible right? The block here doesn't complain if so. This needs to be changed to a match statement to it is exhaustive.

Reply:
Great point -- this got pretty hairy and I should have refactored it, your version looks a lot cleaner